File: KataPython/KataSearching/LinearSearch.py
# Linear Search Algorithm
# Time Complexity: O(n)

"""linearSearch(x, arr) -> int
x: int
arr: list of int
return: int
return the index of x in arr if x is in arr, otherwise return -1
"""
def linearSearch(x, arr):
    for i in range(len(arr)):
        if arr[i] == x:
            return i
    return -1

File: KataJavaScript/LinearSearch.js
// Linear search is the most basic search algorithm. It is a method for finding a target value within a list.
export default function linearSearch(haystack, needle) {
  for (let i = 0; i < haystack.length; i++) {
    if (haystack[i] === needle) {
      return i;
    }
  }
  return -1;
}
